lib/api/objectCopy.js

```diff
@@ -492,6 +492,19 @@ function objectCopy(authInfo, request, sourceBucket,
                         { error: err });
                         return next(err, null, destBucketMD);
                     }
+
+                    const location = destDataGetInfoArr?.[0]?.dataStoreName;
+                    if (location === destBucketMD.getLocationConstraint() && destBucketMD.isIngestionBucket()) {
+                        // If the object is being written to the "ingested" storage location, keep the same
+                        // versionId for consistency and to avoid creating an extra version when it gets
+                        // ingested
+                        // eslint-disable-next-line
+                        const backendVersionId = decodeVID(destDataGetInfoArr[0].dataStoreVersionId);
+                        if (!(backendVersionId instanceof Error)) {
+                            options.versionId = backendVersionId; // eslint-disable-line no-param-reassign
+                        }
+                    }
+
```

Comment:
Copying an object into the same bucket the original object is at is a supported use case. Current behaviour is that it creates a new version of the object. This will introduce a kind of inconsistency depending on wether or not OOB is configured, as now it will rewrite the previous object (or just fail ?). Is this something we know and want ?